conduit/types.py

```diff
@@ -42,3 +44,17 @@ class Stage(Enum):
 LRScheduler: TypeAlias = Union[CosineAnnealingWarmRestarts, ExponentialLR, StepLR]
 MetricDict: TypeAlias = Dict[str, _METRIC_COLLECTION]
 NDArrayR: TypeAlias = Union[npt.NDArray[np.floating], npt.NDArray[np.integer]]
+
+T_co = TypeVar("T_co", covariant=True)
+
+
+@runtime_checkable
+class Sized(Protocol[T_co]):
+    def __len__(self) -> int:
+        ...
```

Comment:
The genericness seems pointless?

Reply:
It's not pointless if you want to make use of genericness in the child classes

Reply:
But then just make the child class generic.

Reply:
That's an option but is there a downside to doing it this way?

Reply:
To me, there seems little reason not to parameterise a protocol and at least give the option to capitalise on it downstream